### ISSUE TITLE
Upgrade to Spring Kafka 2.8.3

### ISSUE THREAD

**Issue by snicoll** [spring-projects/spring-boot#29722]

**Comment by wilkinsona** [comment]
We added support for `CommonErrorHandler` in https://github.com/spring-projects/spring-boot/pull/27927. I think we should just suppress the deprecation warnings while people migrate to `CommonErrorHandler`.

**Comment by garyrussell** [comment]
This change was introduced (in 2.8) to allow a common (auto configured) container factory to be used for both types of listener.

I have now deprecated the setters to help users understand the migration path.

I think we can remove auto configuring the legacy error handlers in 3.0 and, yes, suppress the warning in 2.7.

I have removed all the legacy error handler implementations in 3.0 (and deprecated the legacy interfaces).
